.github/actions/early-terminator/action.yml

```diff
@@ -0,0 +1,39 @@
+name: "early terminate"
+description: |
+  Terminate CI workflow earlier once job failure is detected
+inputs:
+  github-token:
+    description: 'Github token'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Post test results on PR
+      uses: actions/github-script@v4.0.2
+      with:
+        script: |
+            // use github REST apis: https://docs.github.com/en/rest/reference/actions#workflow-runs
+            const https = require('https');
+            const options = {
+              hostname: 'api.github.com',
+              path: `/repos/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}/cancel`,
```

Comment:
We send the signal to cancel the workflow, but would Bors continue to listen for it?  Put another way, do we need a hook to tell Bors?

Reply:
Nope, that's actually the benefit of this approach. If you look at the above tests Zihao did there's a ~10min difference between when bors started the canary and when bors posted that there was a failure (instead of the usual 50+). The biggest question I have is, what happens to LBT when canceled? Is the cluster freed immediately?

Reply:
Bors will report failed as above canary cases

Reply:
💔 Test Failed - ci-test

Reply:
we can treat this approach as the same by clicking the cancel bottom manually, which is using the same rest api to trigger cancellation (I grabbed this api by web source code). Bors will report failed once any job is failed, so the duration is depend on when will be the first job fail, maybe from very start by few secs failed in prepare step or few 10 mins in LBT. @bmwill I think we do not really care about LBT, since LBT is the long tail, most of cases of early termination, LBT even does not be triggered yet. Or if we cancel workflow at the half of LBT runs, Forge cluster will do cleanup as well since cleanup is in destructor. Or for worst case, current LBT does not do cleanup, but next run will do cleanup as well prior the test.

Reply:
@bmwill oh, I think I got your point, your concern is even workflow is cancelled, but cluster still keep running test until finish. I need to double check this case, this is a really good point. But still, this early termination can still gain benefits on cases which have job failed at lint, building-image, unit-test etc